src/asm3/dbupdate.py

```diff
@@ -242,6 +243,26 @@ def fclob(name, nullable = True):
     sql += index("adoption_ReturnedReasonID", "adoption", "ReturnedReasonID")
     sql += index("adoption_ReturnedByOwnerID", "adoption", "ReturnedByOwnerID")
     sql += index("adoption_TrialEndDate", "adoption", "TrialEndDate")
+    sql += index("adoption_AdoptionEventId", "adoption", "AdoptionEventID")
+
+    sql += table("adoptionevent", (
+        fid(),
+        fdate("StartDateTime"),
+        fdate("EndDateTime"),
+        fstr("EventName", True),
+        flongstr("EventDescription", True)))
+    sql += index("adoptionevent_StartDateTime", "adoptionevent", "StartDateTime")
+    sql += index("adoptionevent_EndDateTime", "adoptionevent", "EndDateTime")
+    sql += index("adoptionevent_EventName", "adoptionevent", "EventName")
+
+    sql += table("adoptioneventanimal", (
+        fid(),
+        fint("AdoptionEventID"),
+        fint("AnimalID"),
+        fint("WillReturnToFoster", True),
+        fdate("ArrivalDate")))
```

Comment:
What is the purpose of WillReturnToFoster and ArrivalDate? The adoption event won't take the animal out of the care of the shelter, so presumably if a foster movement was already active you could infer whether it is going back to the fosterer? 

Is arrival date for multi-day events and tracking the animal arriving at the event?

Reply:
These fields are to be used to track the animals in the event - we saw that instead of a boolean you use you use a timestamp to track things (like given vaccination) so arrival date tracks which animal arrived at the event. WillReturnToFoster is used to help the event planner know whether the animal returns to same foster family when the event ends (and the animal wasn't adopted)

Reply:
Ok, I understand ArrivalDate. Couldn't WillReturnToFoster be inferred from animal.ActiveMovementType=2 without needing the column at all? Other than the question over that I'm happy with everything else.

Reply:
@bobintetley if I understand correctly, you suggest to check whether the ReturnDate is null or not on the active Movement (Type=2)? 

What we want to get to with this eventually is a method for the people in the field to easily manage the animals that are brought to the adoption event - we intend to show a list of the animals and allow single item and bulk edit in a similar fashion to the vaccinations list (per animal, or the general vaccinations list). One of the hardships they face is to figure out which animal goes back to its foster family or whether they have to find a new fosterer if the animal is not adopted. I suppose we can use the same mechanism and instead of using this column, set the ReturnDate on the active foster movement if the animal is not returning to its foster family, but in the case the user made a mistake, undoing the ReturnDate update might be harder to accomplish from the simple list UI.

What do you think?

Reply:
I think that's fine. The user shouldn't need to handle returning the existing foster though - when you create a new foster or adoption movement, the backend code will automatically return any existing foster movement.

Ok, I'll do the merge and remove WillReturnToFoster.

Reply:
@bobintetley so how would the user use the list of animals that show up on the event to know which ones need a new foster? The way they do it now is they print a list of animals, and use a pencil to mark which animals return to current foster and which ones don't.

Reply:
If ActiveMovementType=2 then the animal has an active foster to return to. If ActiveMovementType=0 then the animal does not. 

Some users have added their own "Foster Needed" animal flag that they link an emblem to. It means the animal's name shows with a little icon to indicate that it is in need of a fosterer. They can also use the "Flag" mode of shelter view to identify animals needing a fosterer.

Reply:
ActiveMovementType=0 indeed means that there's no active foster, but not all animals need to find a foster (e.g. they return to their original owner, or they stay at the shelter). So within the adoption event management, you think that we'd better use an animal flag to indicate which animal needs to find a foster?

Reply:
Yes, since it will work irrespective of whether or not the animal is at an event.